riscv_executor/src/lib.rs

```diff
@@ -105,36 +105,83 @@ pub enum MemOperationKind {
 
 #[derive(Debug)]
 pub struct MemOperation {
-    /// Line of the register trace the memory operation happened.
-    pub idx: usize,
+    /// The row of the execution trace the memory operation happened.
+    pub row: usize,
     pub kind: MemOperationKind,
     pub address: u32,
 }
 
+pub struct RegWrite {
+    /// The row of the execution trace this write will effect. Multiple writes
```

Comment:
The weird wording was on purpose, because the result of the write will end up on that row. But to use "effect" implies that a row is the consequence of a single write, which is not necessarily true.

Let me think of a better way of writing that.

Reply:
If you want to use the verb, it should be `affect`. The noun, `effect`. That's all I meant.

Reply:
"to effect" is a verb too, and is similar, but a little different from "to affect"

Which is irrelevant now because I changed to neither affect nor effect.